include/seqan3/io/sequence_file/format_genbank.hpp

```diff
@@ -0,0 +1,242 @@
+              typename id_type,
+              typename qual_type>
+    void write(stream_type                        & stream,
+               sequence_file_output_options const & options,
+               seq_type                           && sequence,
+               id_type                            && id,
+               qual_type                          && SEQAN3_DOXYGEN_ONLY(qualities))
+    {
+        std::ranges::ostreambuf_iterator stream_it{stream};
+        size_t sequence_size{0};
+        if constexpr (!detail::decays_to_ignore_v<seq_type>)
+            sequence_size = ranges::size(sequence);
+
+        // ID
+        if constexpr (detail::decays_to_ignore_v<id_type>)
+        {
+            throw std::logic_error{"The ID field may not be set to ignore when writing genbank files."};
+        }
+        else if (ranges::empty(id)) //[[unlikely]]
+        {
+            throw std::runtime_error{"The ID field may not be empty when writing genbank files."};
+        }
+        else if (options.embl_genbank_complete_header)
+        {
+            std::ranges::copy(id, stream_it);
+        }
+        else
+        {
+            std::ranges::copy(std::string_view{"LOCUS       "}, stream_it);
+            std::ranges::copy(id, stream_it);
+            std::ranges::copy(std::string_view{"                 "}, stream_it);
+            std::ranges::copy(std::to_string(sequence_size), stream_it);
+            std::ranges::copy(std::string_view{" bp\n"}, stream_it);
+        }
+
+        // Sequence
+        if constexpr (detail::decays_to_ignore_v<seq_type>) // sequence
+        {
+            throw std::logic_error{"The SEQ field may not be set to ignore when writing genbank files."};
+        }
+        else if (ranges::empty(sequence)) //[[unlikely]]
+        {
+            throw std::runtime_error{"The SEQ field may not be empty when writing genbank files."};
+        }
+        else
+        {
+            std::ranges::copy(std::string_view{"ORIGIN\n"}, stream_it);
+            auto seq = sequence | ranges::view::chunk(60);
+            size_t i = 0;
+            size_t bp = 1;
+            
+            while (bp < sequence_size)
+            {
+                // Sequence length with more than 9 digits are not possible in one genbank entry, maximal 350 kb are
+                // allowed. See: https://www.ncbi.nlm.nih.gov/Sitemap/samplerecord.html#SequenceLengthA
+                for (size_t j = std::to_string(bp).size(); j < 9; j++)
+                    stream_it = ' ';
+                std::ranges::copy(std::to_string(bp), stream_it);
+                stream_it = ' ';
+                std::ranges::copy(seq[i] | view::to_char
+                                         | ranges::view::chunk(10)
+                                         | std::view::join(' '), stream_it);
+                bp += 60;
+                ++i;
+                detail::write_eol(stream_it,false);
+            }
```

Comment:
I am wondering whether one couldn't use two view::interleaves? One that interleaves the newline and one that interleaves the spaces...

Reply:
I am not sure if this make sense, because I need the while loop anyways, so I can either add a newline inside the while loop line by line or after the while loop with an interleave for all lines at once. Is there a performance difference? Because we don't save a line.